src/runtime_src/doc/toc/xrt_kernel_executions.rst

```diff
@@ -0,0 +1,136 @@
+
+**Note:** As noted in the above sequence, the AP_CTRL_CHAIN only applicable when the kernel produces the outputs for the pending requests in-order. Kernel servicing the requests out-of-order cannot be supported by AP_CTRL_CHAIN execution model. 
+
+**Output synchronization**
+
+1. The XRT driver waits for ap_done to be asserted by the kernel (guaranteeing the output data is fully produced by the kernel).
+2. The XRT driver writes a 1 in ap_continue to keep kernel running 
+
+The input and output synchronization occurs asynchronously, as a result, multiple executions are performed by the kernel in an overlapping or pipelined fashion.
+
+Control Signal Topology
+-----------------------
+The signals ap_start, ap_ready, ap_done, ap_continue must be connected to the AXI_LITE control and status register (at the address 0x0 of the AXI4-Lite Slave interface) section to specific bits.  
+
+====== ===================== =======================================================================
+ Bit       Signal name                                  Description
+====== ===================== =======================================================================
+  0         ap_start           Asserted by the XRT when kernel can process the new data
+  1         ap_done            Asserted by the kernel when it is finished producing the output data
+  3         ap_ready           Asserted by the kernel when it is ready to accept the new data
+  4         ap_continue        Asserted by the XRT to allow kernel keep running
+====== ===================== =======================================================================
+
+Host Code Consideration
+-----------------------
+The host code exercising a AP_CTRL_CHAIN kernel should be able to fill the input queue with multiple execution requests well ahead to take the advantage of pipelined nature of the kernel. For example, considering OpenCL host code, it should use out-of-order command queue for multiple kernel execution requests. The host code should also use API *clEnqueueMigrateMemObjects* to explicitly migrate the buffer before the kernel execution. 
+
+
+
+=====================================
+AP_CTRL_NONE (Continuously Running Kernel)
+=====================================
+
+Sometimes the kernel does not need to be controlled by the host. For example, if the kernel is only communicating through the stream, it only works when the data is available at its input through the stream, and the kernel stalls when there is no data to process, waiting for new data to arrive through the stream to start working again. These type of kernels has no control signal connected to the AXI4-Lite Slave interface. 
+
+**Important points to remember**
+
+1. Consider a kernel with AP_CTRL_NONE only when it has no Memory input and output.
+2. There is no need to start the kernel by *clEnqueueTask* or *clEnqueueNDRangeKernel* from the host. 
```

Comment:
*clEnqueueTask* → `clEnqueueTask` and so on, since it is an API and not a concept